pkg/server/apiHandler.go

```diff
@@ -301,6 +304,65 @@ func (h *apiHandler) logs(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusCreated)
 }
 
+func (h *apiHandler) newLogs(w http.ResponseWriter, r *http.Request) {
+	cert := getClientCert(r)
+	u, err := h.manager.DeviceCheckCert(cert)
+	if err != nil {
+		log.Printf("error checking device cert: %v", err)
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return
+	}
+	h.recordClient(u, r)
+	gr, err := gzip.NewReader(r.Body)
+	if err != nil {
+		log.Printf("error gzip.NewReader: %v", err)
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return
+	}
+	logBundleFromHeader := &logs.LogBundle{}
+	if err := json.Unmarshal([]byte(gr.Comment), logBundleFromHeader); err != nil {
```

Comment:
I am pretty sure that [gzip.Reader](https://golang.org/pkg/compress/gzip/#Reader) has a property [Header](https://golang.org/pkg/compress/gzip/#Header) which in turn has a property `Comment`, so `gr.Comment` should be incorrect, but `gr.Header.Comment` should work.

Reply:
Hmm, I thing, they have the same value. You can see the example on top of godoc [gzip](https://golang.org/pkg/compress/gzip) (inside the Overview section).

Reply:
Ah, you are right. It is an embedded struct. I missed that. Never mind then, all good. 🤦🏻‍♂️